Feature #2772 Apptainer documentation (#2783)
dtcenter/METplus: docs/Users_Guide/getting_started.rst

Please note that use cases that have additional Python package dependencies
may not run successfully unless those packages are installed inside the
container or obtained elsewhere.

.. _metplus-apptainer:

METplus in Apptainer
--------------------

Apptainer (formerly Singularity) can also be used to run METplus.

> Images can be pulled from DockerHub using Apptainer commands.

Reviewer: ```suggestion
Images can be pulled from DockerHub using Apptainer commands.

If running on the NCAR Casper cluster, be sure to first load the Apptainer module via::

    module load apptainer
```

Navigate to a working directory and pull an image from DockerHub, e.g.::

    apptainer pull docker://dtcenter/metplus:5.1-latest

This will create a *.sif* file in the current directory,
e.g. **metplus_5.1-latest.sif**.

See :ref:`metplus-docker` for information about available images on DockerHub.

.. _apptainer-sample-input:

Sample Input Data
^^^^^^^^^^^^^^^^^

> The sample input data Docker data volumes that are provided on DockerHub are

Reviewer: ```suggestion
The Docker data volumes that contain sample input data provided on DockerHub are
```

> not compatible with Apptainer. Sample input data are also provided on the web.

Reviewer: ```suggestion
not compatible with Apptainer. Therefore, sample input data must be obtained from from the METplus Data website.
```

Navigate to the `METplus Data <https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data>`_

> page. Next, navigate to the directory that corresponds to the vX.Y version that

Reviewer: ```suggestion
website. Next, navigate to the directory that corresponds to the vX.Y version that
```

will be run, e.g. `v5.1 <https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v5.1>`_
The names of sample data tar files include the corresponding use case category.

Download one or more of the sample data tar files and uncompress them into
a local directory. The directory that contains the uncompressed sample data

> will be mounted to the container to make the data available inside it.

Reviewer: ```suggestion
will need to be mounted to the container to make the data available inside it.
```

    export APPTAINER_BIND="./input_data:/data/input/METplus_Data"

Running METplus in Apptainer
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

The METplus wrappers inside the container come pre-configured so that
:ref:`sys_conf_input_base` is set to /data/input/METplus_Data.
Mounting the local data directory to this path will allow the use cases
to be run without the need to modify the default configuration.

The :ref:`sys_conf_output_base` value is set to /data/output by default.
This directory can also be mounted to a local directory so the output from
a use case run can be accessed locally. Multiple directory mounts can be specified
by separating them by comma::

    export APPTAINER_BIND="./input_data:/data/input/METplus_Data,./output_data:/data/output"

In this example, the directory **./output_data** will be populated with the
output from the use case.

To run a use case, run the **apptainer exec** command with the **.sif**
file that was created by the **apptainer pull** command and the call to the
**run_metplus.py** script as arguments::

    apptainer exec metplus_5.1-latest.sif /metplus/METplus/ush/run_metplus.py /metplus/METplus/parm/use_cases/met_tool_wrapper/GridStat/GridStat.conf

> Note that the full path to the run_metplus.py script must be provided.

Reviewer: ```suggestion
Note that the full path to the run_metplus.py script and the configuration file for the use case you are running must be provided.
```
